lib/fortitude/widget/caching.rb

```diff
@@ -0,0 +1,38 @@
+module Fortitude
+  class Widget
+    module Caching
+      extend ActiveSupport::Concern
+      include ActionView::Helpers::CacheHelper
+
+      module ClassMethods
+        # PUBLIC API
+        def cacheable(opts = {})
+          if extra_assigns == :use
+            extra_assigns :ignore
+          end
```

Comment:
We don't want to use extra assigns when calculating the cache key, since this would make the cache key be too variable. (`@marked_for_same_origin_verification` was getting set in my Rails app, for example.) However, this means that the developer cannot use extra assigns in their widget. This is a bit ugly... maybe we could raise an error if the developer tries to access something in `extra_assigns` instead?